src/coreclr/jit/morphblock.cpp

```diff
@@ -1571,9 +1571,8 @@ GenTree* Compiler::fgMorphStoreDynBlock(GenTreeStoreDynBlk* tree)
     if (tree->gtDynamicSize->IsIntegralConst())
     {
         int64_t size = tree->gtDynamicSize->AsIntConCommon()->IntegralValue();
-        assert(FitsIn<int32_t>(size));
 
-        if (size != 0)
+        if ((size != 0) && FitsIn<int32_t>(size))
```

Comment:
If i understand correctly, with your casting changes in importer, we would hit this assert and hence moving this to if-check is necessary?

Reply:
We should not see this in practice since we only create these nodes in very limited circumstances, when we see `initblk`/`cpblk`. But it wouldn't be considered illegal to have a `CNS_INT` node in this position with a native int > 2^31 anymore, so we shouldn't have asserts about that.